jorge/js/app.js

```diff
@@ -1,20 +1,134 @@
-//Variables
-const register = document.getElementById('register')
-const login = document.querySelector('#login')
+const APP_BUTTONS = {
+  NAV_LOGIN: document.querySelector("#navLogin"),
+  NAV_SIGN_UP: document.querySelector("#navSignUp"),
+  SHOW_LOGIN: document.querySelector("#showLogin"),
+  SHOW_SIGN_UP: document.querySelector("#showSignUp"),
+  LOGIN_CANCEL: document.querySelector("#loginCancel"),
+  SIGN_UP_CANCEL: document.querySelector("#signUpCancel"),
+  SUBMIT_LOGIN: document.querySelector("#submitLogin"),
+  SUBMIT_SIGN_UP: document.querySelector("#submitSignUp"),
+};
 
-// addEventListeners
+const APP_SECTIONS = {
+  WELCOME_WRAPPER: document.querySelector("#welcomeWrapper"),
+  LOGIN_WRAPPER: document.querySelector("#loginWrapper"),
+  SIGN_UP_WRAPPER: document.querySelector("#signUpWrapper"),
+  MOVEMENTS_WRAPPER: document.querySelector("#movementsWrapper"),
+};
 
-register.addEventListener('click', showRegisterForm)
+const APP_FORMS = {
+  LOGIN_FORM: document.querySelector("#loginForm"),
+  SIGN_UP_FORM: document.querySelector("#signUpForm"),
+};
 
-login.addEventListener('click', showLoginForm)
+const APP_SPAN = {
+  USER_NAME: document.querySelector("#userLoggedName"),
+  USER_BALANCE: document.querySelector('#userBalance')
+}
+
+function toggleVisibility(...elements) {
+  elements.forEach((element) => {
+    const isElementHidden = element.hasAttribute("hidden");
+    isElementHidden
+      ? element.removeAttribute("hidden")
+      : element.setAttribute("hidden", "");
+  });
+}
+
+function setViewSection(elementViewed) {
+  for (const section in APP_SECTIONS) {
+    APP_SECTIONS[section] === elementViewed
+      ? APP_SECTIONS[section].removeAttribute("hidden")
+      : APP_SECTIONS[section].setAttribute("hidden", "");
+  }
+}
+
+function setClickHandlers(callback, ...buttons) {
+  buttons.forEach((button) => button.addEventListener("click", callback));
+}
 
-// Functions
+function parseFormData(form) {
+  const data = {};
+  const formData = new FormData(form);
+  for (let formFields of formData) {
+    data[formFields[0]] = formFields[1];
+  }
+  return data;
+}
 
-function showRegisterForm () {
-  console.log('SHOW REGISTER')
+function submitSignUp(event) {
+  event.preventDefault();
+  const userData = parseFormData(APP_FORMS.SIGN_UP_FORM);
+  userData.balance = 0
+  localStorage.setItem(userData.username, JSON.stringify(userData));
+  toggleVisibility(APP_SECTIONS.LOGIN_WRAPPER, APP_SECTIONS.SIGN_UP_WRAPPER)
 }
 
-function showLoginForm () {
-  console.log('SHOW LOGIN')
+function login(e) {
+  e.preventDefault();
+  const loginData = parseFormData(APP_FORMS.LOGIN_FORM);
+  const signUpData = JSON.parse(localStorage.getItem(loginData.username))
+
+  loginData.balance = signUpData.balance
+
+  allowLogin(loginData, signUpData)
+
+}
+
+function allowLogin (loginData, signUpData) {
+  if (loginData.username === signUpData.username && loginData.password === signUpData.password) {
```

Comment:
NIT: extract this into a variable like `const isUserRegistered = loginData.username === signUpData.username && loginData.password === signUpData.password; if(isUserRegistered){...}`